src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.service.producers.BulldozerProducer;
+import core.mate.academy.service.producers.ExcavatorProducer;
+import core.mate.academy.service.producers.TruckProducer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer<? extends Machine> producer = null;
+        if (type == Bulldozer.class) {
+            producer = new BulldozerProducer();
+        } else if (type == Excavator.class) {
+            producer = new ExcavatorProducer();
+        } else if (type == Truck.class) {
+            producer = new TruckProducer();
+        } else {
+            return new ArrayList<>();
+        }
+        return (List<Machine>) producer.get();
```

Comment:
You do not need to cast it, `get()` method should already return List<Excavator> or other model, which is already a `Machine` instance
```suggestion
        return producer.get();
```